File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int temp = 0;
        int rows = matrix.size();
        int columns = matrix[0].size();


        for (int i = 0; i < rows; i++) {
            for (int j = 0; j <= i; j++) {
                temp = matrix[j][i];
                matrix[j][i] = matrix[i][j];
                matrix[i][j] = temp;
            }
        }

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns / 2; j++) {
                temp = matrix[i][j];
                matrix[i][j] = matrix[i][columns - 1 - j];
                matrix[i][columns - 1 - j] = temp;
            }
        }

    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        temp: int
        width: int = len(matrix[0])
        height: int = len(matrix)

        for i in range(height):
            for j in range(i, width):
                temp = matrix[i][j]
                matrix[i][j] = matrix[j][i]
                matrix[j][i] = temp
        for i in range(height):
            for j in range(width//2):
                temp = matrix[i][j]
                matrix[i][j] = matrix[i][width-j-1]
                matrix[i][width-j-1] = temp
